watson-developer-cloud/discovery-components
# chore(eslint): disable formatOnSave and add jsx-a11y

#### What do these changes do/fix?
This disables the formatOnSave hook as well as splits up our lint rules for js,jsx,ts,tsx and other files in our repo. This also enables the `jsx-a11y` plugin for issue https://github.ibm.com/Watson-Discovery/disco-widgets/issues/166 and https://github.ibm.com/Watson-Discovery/disco-widgets/issues/168 . I'm opening this as a draft PR to try to get some feedback from the doc-viz team. When enabling the `jsx-a11y` plugin, the `Section.tsx` component contains some violations. Feedback from the owners of this component would help me more confidently tackle these violations as the fix requires adding event listeners ect (more detail contained in the diff comments)


#### How do you test/verify these changes?
- Make sure that the formatOnSave hook does not run
- Make sure that prettier is run on the pre-commit hook
- Make sure that the a11y plugin is added and that there are no violations

#### Have you documented your changes (if necessary)?
N/A

#### Are there any breaking changes included in this pull request?
no
